includes/classes/Jobber.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Jobber API Connection
+ *
+ * @package Jobber
+ */
+
+namespace Jobber;
+
+use WP_Error;
+use TenupFramework\Module;
+
+/**
+ * Base class for Jobber API Connection
+ */
+class Jobber {
+	use Module;
+
+	/**
+	 * API URL
+	 *
+	 * @var string
+	 */
+	protected $api_url = 'https://api.getjobber.com/api';
+
+	/**
+	 * API Access Token
+	 *
+	 * @var string
+	 */
+	private $access_token;
+
+	/**
+	 * Refresh Token
+	 *
+	 * @var string
+	 */
+	private $refresh_token;
+
+	/**
+	 * Module Constructor
+	 */
+	public function __construct() {
+		$this->access_token  = Auth::get_token( 'access' );
+		$this->refresh_token = Auth::get_token( 'refresh' );
+	}
+
+	/**
+	 * Can we register this module?
+	 *
+	 * @return boolean
+	 */
+	public function can_register() {
+		return true;
+	}
+
+	/**
+	 * Hook module into WP.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		add_filter( 'allowed_redirect_hosts', [ $this, 'allow_jobber_redirect' ] );
+	}
+
+	/**
+	 * Add the middleware URL to the allowed redirect hosts.
+	 *
+	 * @param array $hosts Allowed Redirect Hosts.
+	 * @return array
+	 */
+	public function allow_jobber_redirect( $hosts ) {
+		$hosts[] = wp_parse_url( Auth::$url, PHP_URL_HOST );
+		return $hosts;
+	}
+
+	/**
+	 * Execute a GraphQL Query.
+	 * We must use cURL because wp_remote_post does not work with GraphQL.
+	 *
+	 * @param string $query GraphQL Query.
+	 * @return array|WP_Error
+	 */
+	protected function query( $query ) {
+		if ( empty( $this->access_token ) ) {
+			return new WP_Error( 'jobber_no_access_token', 'No access token found.' );
```

Comment:
Not sure if we output this error message anywhere but still might be ideal to pass it through translation